frontend/src/components/dimensions/models.ts

```diff
@@ -0,0 +1,20 @@
+// | null | undefined to make it easier to work with GraphQL
```

Comment:
`x?: string | null` isn't exactly `x: string | null | undefined` – with `?` you can elide the field, with the other you need to pass it, even if the value is `undefined`. This is useful knowledge for some prop types where you want the caller to pinky-swear they know what they're doing.

Reply:
In this instance, `title` is usually present unless the user chose not to include it in the GraphQL field list, and `color` may not be present if the use case of dimensions does not provide dimension colors. So I'd say `x?: string | null` is fairly close to the truth here.